openvdb/points/AttributeArray.cc

```diff
@@ -52,81 +52,71 @@ struct LockedAttributeRegistry
     AttributeFactoryMap mMap;
 };
 
-
-// Declare this at file scope to ensure thread-safe initialization.
-tbb::spin_mutex sInitAttributeRegistryMutex;
-
-
 // Global function for accessing the registry
 LockedAttributeRegistry*
 getAttributeRegistry()
 {
-    tbb::spin_mutex::scoped_lock lock(sInitAttributeRegistryMutex);
-
-    static LockedAttributeRegistry* registry = nullptr;
+    static LockedAttributeRegistry registry;
+    return &registry;
```

Comment:
@mortenbh - I took this opportunity to try and incorporate your contribution to resolve this static pointer memory leak, thanks!

Reply:
@danrbailey That's great! Any chance I could work with you to incorporate the rest of my changes as well? :)

Reply:
We discussed this in the last TSC meeting and I believe Jeff has replied to your PR now. :)